### PR TITLE
agent: stop trailing chatter from clobbering a subagent's <review> result

### PR DESCRIPTION
**Pull request by devxoul**
## Problem

A `reviewer` subagent produced its `<review>` block correctly, but the parent's `subagent_output` returned only a useless summary — so the parent re-spawned the reviewer, wasting a `deep`-model run.

Root cause is two coupled bugs:

1. **The reviewer called the wrong tool name** (`web_search` instead of the real `websearch`). That triggered `attachToolNotFoundNudge`, which `steer`s a "did you mean" reminder into the running turn.
2. **The nudge had no dedup** — it re-fired on every errored `tool_execution_end`. A wedged model emitted a short "review delivered above" turn per reminder.
3. **`attachFinalMessageCapture` took the *last* assistant `message_end` as `finalMessage`** — so each trailing summary overwrote the real `<review>` block. (It also didn't filter on role, so a `user`/`toolResult` echo could clobber the answer too.)

Observed sequence (production logs): reviewer emits full `<review>` → nudge fires → 225-tok summary overwrites it → nudge fires again → 165-tok summary overwrites again → parent receives the summary, re-spawns.

## Fix

Two coupled, minimally-scoped changes:

- **`attachToolNotFoundNudge`** — nudge **once per missing tool name** (`Set` guard). A model stuck on a wrong name no longer manufactures extra assistant turns.
- **`attachFinalMessageCapture`** — prefer the **last complete `<review>…</review>` block** over a later plain message, and **ignore non-assistant `message_end`** events. Falls back to the last assistant message when no block is present, so free-form subagents (explorer/scout/operator) are unaffected.

The capture fix is scoped to the reviewer's existing `<review>` contract rather than introducing a general `SubagentOutputContract` abstraction — that broader refactor is deferred until the same clobbering class shows up in a free-form subagent.

## Edge cases covered by tests

- Full `<review>` followed by a trailing summary → returns the review (the production regression).
- Two `<review>` blocks → returns the **later** one (revision, not stale).
- Bare/unclosed `<review>` literal inside reviewed text → **not** treated as a block (non-greedy, requires close tag).
- No `<review>` block → falls back to the last assistant message.
- Non-assistant (`toolResult`) `message_end` → does not clobber the assistant answer.
- Nudge fires **once** per bad tool name; still fires for a *different* bad name.

## Verification

- `bun run typecheck` (exit 0)
- `bun run lint` (0 errors; 66 pre-existing warnings, none in changed files)
- `bun run format:check` (clean)
- `bun run test` (7017 pass; the single `resolveSelfPackageJsonPath` failure is environmental — it asserts the checkout dir is named `typeclaw`, but the worktree is `typeclaw-review-capture`; verified it also fails on clean `origin/main` in this worktree and will pass in CI)

### CODE REVIEW

File: src/agent/subagents.ts [hunk 1]
```diff
@@ -403,13 +403,29 @@ function raceSubagentCompletion(
...
       if (ev?.type !== 'message_end') return
+      // Real assistant messages carry role 'assistant'; older test doubles omit
+      // it. user/toolResult echoes must never overwrite the assistant's answer.
+      const role = ev.message?.role
+      if (role !== undefined && role !== 'assistant') return
       const text = extractFinalMessageText(ev.message?.content)
-      if (text !== null) onFinalMessage(text)
+      if (text === null) return
+      lastAssistant = text
+      if (REVIEW_BLOCK_RE.test(text)) lastReview = text
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/agent/subagents.ts]: Good catch — fixed in 37b1e606. The capture now extracts the matched block via `text.match(REVIEW_BLOCK_RE)` and stores that substring, not the whole assistant text. The regex is global and I keep the **last** match, so a single message containing preamble + block + trailing chatter returns just the block, and multiple blocks in one message yield the revision. Added two tests: same-message preamble/trailing chatter is stripped, and last-of-multiple-blocks wins.
